### PR TITLE
181889832 V3: attribute edit properties

### PR DESCRIPTION
**Pull request by eireland**
Adds Edit Attribute Properties modal where user can edit the attribute properties:
Name: changes the attribute name in the case table
Description: adds a description to the attribute. Description appears as tooltips when column header is hovered.
Type: allows users to select different types but does nothing
Unit: allows users to enter units for the attribute. Unit appears in parenthesis in the column header
Editable: allows users to select true or false but does nothing yet

### CODE REVIEW

**Comment by bfinzer** [comment]
I notice that
* The Cancel button acts like Apply in that the original values are replaced with new ones
* The **type** of an attribute seems to always default to *numeric*. If the user has not specified a type, then the default should be empty. But a nice **touchup** for V3 would be to display *none* instead of empty and to include *none* in the menu.

**Review by bfinzer — APPROVED** [review]
👍🏻LGTM
There are some needed improvements to behavior, but they need not hold things up.

File: v3/src/components/case-table/edit-attribute-properties.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      <FormLabel display="flex" flexDirection="row">precision:
+        <NumberInput size="xs" min={0} ml={5} data-testid="attr-precision-input" value={precision}
+          onFocus={(e) => e.target.select()} onChange={value => onPrecisionChange(value)}
+          onKeyDown={(e)=>e.stopPropagation()}
+        >
+          <NumberInputField placeholder="precision" />
+          {/* TODO: There's a weird behavior in the ui that when you click on the stepper, the number input element
+                    loses focus and the first input element in the modal gets the focus. Seems like the modal is getting
+                    blurred and refocused when you click on the stepper. And when the modal is refocused, it focuses on
+                    the first input element in the modal */}
```
[the remark this answers is not in the document]
Comment (eireland) [reply v3/src/components/case-table/edit-attribute-properties.tsx]: That makes things easier!

**Review by kswenson — APPROVED** [review]
👍 Looks good -- sorry for the delay in getting back to you. There were a couple of things I wanted to check and in the end I decided to just add a commit and then merge master into the branch rather than go back and forth with code review comments. Take a look and make sure I haven't broken anything and then feel free to merge.

**Pull request merged**